### PR TITLE
Android Documentation

### PR DESCRIPTION
**Pull request by igor-egorov**
### Description of the Change

PR introduces documentation about Android Client Library.

### Benefits

Documentation answers the following set of questions for Android developer: where to get, how to build, how to use, where to find code examples.

### Possible Drawbacks 

The guide presented in this PR is not relevant until #1133 and #1073 are merged into develop branch.

### Usage Examples or Tests *[optional]*

If you want to test the guide, then you have to download android-build.sh script from #1073 and change the line 
``git clone -b develop --depth=1 https://github.com/hyperledger/iroha`` 
to 
``git clone -b fix/swig-patch-path --depth=1 https://github.com/hyperledger/iroha``

### CODE REVIEW

File: docs/source/guides/libraries.rst [hunk 1]
```diff
@@ -1,91 +1,314 @@
...
+    Please `download <https://developer.android.com/ndk/downloads/index.html>`__ and unpack NDK to any suitable folder.
+
+automake
+    .. parsed-literal::
+        sudo apt install automake
+        automake --version
+        # automake (GNU automake) 1.15
+
+bison
+    .. parsed-literal::
```
Comment (neewy) [inline docs/source/guides/libraries.rst]: .. code-block:: shell

File: docs/source/guides/libraries.rst [hunk 2]
```diff
@@ -1,91 +1,314 @@
...
+The guide was tested on systems running Ubuntu 16.04 and macOS.
+
+Prerequisites
+"""""""""""""
+
+Android NDK
+    Please `download <https://developer.android.com/ndk/downloads/index.html>`__ and unpack NDK to any suitable folder.
+
+automake
+    .. parsed-literal::
```
Comment (neewy) [inline docs/source/guides/libraries.rst]: .. code-block:: shell

File: docs/source/guides/libraries.rst [hunk 3]
```diff
@@ -1,91 +1,314 @@
...
+        bison --version
+        # bison (GNU Bison) 3.0.4
+
+cmake
+    Minimum required version is 3.8, but we recommend to install the latest available version (3.10.3 at the moment).
+    
+    Since Ubuntu repositories contain unsuitable version of cmake, you need to install the new one manually.
+    Here is how to build and install cmake from sources.
+
+    .. parsed-literal::
```
Comment (neewy) [inline docs/source/guides/libraries.rst]: .. code-block:: shell

File: docs/source/guides/libraries.rst [hunk 4]
```diff
@@ -1,91 +1,314 @@
...
+
+`Android API levels <https://developer.android.com/guide/topics/manifest/uses-sdk-element.html#ApiLevels>`__
+
+Please use the same root part of Java package name for library build as you use for your Android project.
+For example, your project is located in a package called ``com.mycompany.androidapp``, so please consider to build the library in a
+package, which name starts with ``com.mycompany`` (e.g. ``com.mycompany.iroha``).
+
+A couple of launch commands examples:
+
+.. parsed-literal::
```
Comment (neewy) [inline docs/source/guides/libraries.rst]: .. code-block:: shell

File: docs/source/guides/libraries.rst [hunk 5]
```diff
@@ -1,91 +1,314 @@
...
+    # build Java bindings and binary library for arm64-v8a in Release mode
+    ./android-build.sh arm64-v8a 27 /home/user/lib/android-ndk-r16b com.mycompany.iroha
+
+    # build Java bindings and binary library for x86 in Debug mode
+    ./android-build.sh x86 27 /home/user/lib/android-ndk-r16b com.mycompany.iroha Debug
+
+Build artefacts will be collected in ``lib`` directory near the script ``android-build.sh``.
```
Comment (neewy) [inline docs/source/guides/libraries.rst]: artifacts?

File: docs/source/guides/libraries.rst [hunk 6]
```diff
@@ -1,91 +1,314 @@
...
+Importing the Library from jcenter
+""""""""""""""""""""""""""""""""""
+
+The easiest way to use Irohalib for Android is to import the library dependency from `jcenter <https://bintray.com/bulatmukhutdinov/maven/iroha-android-bindings>`__.
+
+All you need to do is a simple set of four steps:
+
+1. Add to your ``build.gradle`` file the following line:
+
+   .. parsed-literal::
```
Comment (neewy) [inline docs/source/guides/libraries.rst]: .. code-block:: groovy

File: docs/source/guides/libraries.rst [hunk 7]
```diff
@@ -1,91 +1,314 @@
...
+       compile 'jp.co.soramitsu.iroha.android:iroha-android-bindings:1.0'
+
+2. Copy the latest version of ``*.proto`` files from ``develop`` branch of Iroha `repository <https://github.com/hyperledger/iroha/tree/develop/schema>`__ into 
+   ``app/src/main/proto/`` folder inside your project in Android Studio.
+
+   The resulting directory structure should look like as follows:
+
+   .. parsed-literal::
```
Comment (neewy) [inline docs/source/guides/libraries.rst]: .. code-block:: shell

File: docs/source/guides/libraries.rst [hunk 8]
```diff
@@ -1,91 +1,314 @@
...
+                    ├── primitive.proto
+                    ├── proposal.proto
+                    ├── queries.proto
+                    ├── responses.proto
+                    └── yac.proto
+
+
+3. Create additional directories ``app/src/main/proto/google/protobuf/`` and place there a file called ``empty.proto`` with the following contents:
+
+   .. parsed-literal::
```
Comment (neewy) [inline docs/source/guides/libraries.rst]: .. code-block:: proto

File: docs/source/guides/libraries.rst [hunk 9]
```diff
@@ -1,91 +1,314 @@
...
+       option java_package = "com.google.protobuf";
+       option java_outer_classname = "EmptyProto";
+       option java_multiple_files = true;
+
+       message Empty {
+       }
+
+4. Add ``protobuf`` and ``grpc`` dependecies and protobuf configuration block into your ``buld.gradle`` file.
+
+   .. parsed-literal::
```
Comment (neewy) [inline docs/source/guides/libraries.rst]: .. code-block:: groovy

File: docs/source/guides/libraries.rst [hunk 10]
```diff
@@ -1,91 +1,314 @@
...
+---------------
+
+The library, in essence, is a set of Java interfaces and binary libraries compiled for different architectures. Supported architectures are *arm, x86, x86_64*.
+
+Where to Get
+^^^^^^^^^^^^
+
+There are two ways to get Iroha library for Android:
+
+#. Include into a project prebuilt library from jcenter repository https://bintray.com/bulatmukhutdinov/maven/iroha-android-bindings.
```
Comment (BulatMukhutdinov) [inline docs/source/guides/libraries.rst]: just add implementation 'jp.co.soramitsu.iroha.android:iroha-android-bindings:1.0'
No need to provide a link to the repo in bintray

File: docs/source/guides/libraries.rst [hunk 11]
```diff
@@ -1,91 +1,314 @@
...
+| 5        | No       | **Build Type**        | Defines build mode of binary part           | ``Debug`` or ``Release``            |
+|          |          |                       | of the library. ``Release`` is the default  |                                     |
+|          |          |                       | option.                                     |                                     |
++----------+----------+-----------------------+---------------------------------------------+-------------------------------------+
+
+`Android API levels <https://developer.android.com/guide/topics/manifest/uses-sdk-element.html#ApiLevels>`__
+
+Please use the same root part of Java package name for library build as you use for your Android project.
+For example, your project is located in a package called ``com.mycompany.androidapp``, so please consider to build the library in a
+package, which name starts with ``com.mycompany`` (e.g. ``com.mycompany.iroha``).
```
Comment (BulatMukhutdinov) [inline docs/source/guides/libraries.rst]: should be com.mycompany.androidapp.iroha

File: docs/source/guides/libraries.rst (hunk 5 above)
Comment (igor-egorov) [reply docs/source/guides/libraries.rst]: Both are acceptable. Artefact in British English and artifact in US English.

Comment (igor-egorov) [reply docs/source/guides/libraries.rst]: Please confirm if you insist on the change.

**Pull request merged**